  2  LatinIME: correctly handle jni fallbacks

The way it is done till now does not make really sense. If
the gesture lib was flashed with gapps from the user it still returns
the value config_gesture_input_enabled_by_build_config back which returns
false by default on aosp

So do not rely on config_gesture_input_enabled_by_build_config at all and only
decide on the type of lib which was loaded.

diff --git a/java/src/com/android/inputmethod/latin/settings/Settings.java b/java/src/com/android/inputmethod/latin/settings/Settings.java
index 188f621..bb89d10 100644
--- a/java/src/com/android/inputmethod/latin/settings/Settings.java
+++ b/java/src/com/android/inputmethod/latin/settings/Settings.java
@@ -231,10 +231,7 @@ public static boolean readBlockPotentiallyOffensive(final SharedPreferences pref
     }
 
     public static boolean readFromBuildConfigIfGestureInputEnabled(final Resources res) {
-        if (!JniUtils.mHaveGestureLib) {
-            return false;
-        }
-        return res.getBoolean(R.bool.config_gesture_input_enabled_by_build_config);
+        return JniUtils.mHaveGestureLib;
     }
 
     public static boolean readGestureInputEnabled(final SharedPreferences prefs,
diff --git a/java/src/com/android/inputmethod/latin/utils/JniUtils.java b/java/src/com/android/inputmethod/latin/utils/JniUtils.java
index 188f622..bb89d10 100644
--- a/java/src/com/android/inputmethod/latin/utils/JniUtils.java
+++ b/java/src/com/android/inputmethod/latin/utils/JniUtils.java
@@ -29,6 +29,8 @@
             System.loadLibrary(JniLibName.JNI_LIB_NAME2);
             mHaveGestureLib = true;
         } catch (UnsatisfiedLinkError ue) {
+            Log.e(TAG, "Could not load native gesture library " + JniLibName.JNI_LIB_NAME2
+                    + ". Loading now native default library ", ue);
             try {
                 System.loadLibrary(JniLibName.JNI_LIB_NAME);
             } catch (UnsatisfiedLinkError ule) {
diff --git a/java/res/values/gesture-input.xml b/java/res/values/gesture-input.xml
index 188f623..bb89d10 100644
--- a/java/res/values/gesture-input.xml
+++ /dev/null
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
-/*
-**
-** Copyright 2012, The Android Open Source Project
-**
-** Licensed under the Apache License, Version 2.0 (the "License");
-** you may not use this file except in compliance with the License.
-** You may obtain a copy of the License at
-**
-**     http://www.apache.org/licenses/LICENSE-2.0
-**
-** Unless required by applicable law or agreed to in writing, software
-** distributed under the License is distributed on an "AS IS" BASIS,
-** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-** See the License for the specific language governing permissions and
-** limitations under the License.
-*/
--->
-<resources>
-    <bool name="config_gesture_input_enabled_by_build_config">false</bool>
-</resources>
